### PR TITLE
Fixes

### PR DESCRIPTION
**Pull request by cittyinthecloud**
All routes work except download, which I think is caused by me being on win32.

### CODE REVIEW

File: framework/objects.py [hunk 1]
```diff
@@ -45,4 +46,5 @@
 auth_service = Authenticator(SETTINGS)
 
 database_handle = DBHandler(user=SETTINGS["DB_USER"], password=SETTINGS["DB_PASS"],
```
Comment (sr229) [inline framework/objects.py:49]: we prefix os environments with `SAYONIKA_`. Please do that.

**Review by sr229 — CHANGES_REQUESTED** [review]
just change this one and it'll be LGTM

File: framework/objects.py (hunk 1 above)
Comment (cittyinthecloud) [reply framework/objects.py:49]: @sr229 it's automatically prefixed earlier

File: framework/authentication.py [hunk 2]
```diff
@@ -37,22 +40,20 @@ def has_authorized_access(self, resource_id: str = None) -> bool:
...
-                       for collection in [
-                               # User can edit their own mods
-                               (mod.id for mod in user.mods),
-                               # User can edit their own reviews
-                               (review.id for review in user.reviews),
-                               # User can edit themselves
-                               (user.id,)
-                       ]):
+            if "mod_id" in kwargs:
+                if kwargs["mod_id"] in (mod.id for mod in user.mods):
```
Comment (Martmists-GH) [inline framework/authentication.py]: Use a negation here and leave `return True` to the end of the function instead.

File: framework/authentication.py [hunk 3]
```diff
@@ -37,22 +40,20 @@ def has_authorized_access(self, resource_id: str = None) -> bool:
...
+                    return True
                 abort(403, "User does not have permission to access this resource")
-
+            elif "user_id" in kwargs:
+                if kwargs["user_id"] == user.id:
```
Comment (Martmists-GH) [inline framework/authentication.py]: Ditto

File: framework/objects.py [hunk 4]
```diff
@@ -45,4 +46,5 @@
...
-                            database="Sayonika", host=SETTINGS["DB_HOST"])
+                            database="sayonika", host=SETTINGS["DB_HOST"],
```
Comment (Martmists-GH) [inline framework/objects.py]: We should add an ENV var for this

**Review by Martmists-GH — CHANGES_REQUESTED** [review]
Seems good overall, make sure it passes snekchek. (Install from izunadevs/snekchek, pip version is broken)

**Pull request merged**